kusto/ingest/internal/properties/properties.go

```diff
@@ -52,117 +55,161 @@ const (
 type DataFormat int
 
+const (
+	// DFUnknown indicates the EncodingType is not set.
+	DFUnknown DataFormat = 0
+	// AVRO indicates the source is encoded in Apache Avro format.
+	AVRO DataFormat = 1
+	// ApacheAVRO indicates the source is encoded in Apache avro2json format.
+	ApacheAVRO DataFormat = 2
+	// CSV indicates the source is encoded in comma seperated values.
+	CSV DataFormat = 3
+	// JSON indicates the source is encoded as one or more lines, each containing a record in Javscript Object Notation.
+	JSON DataFormat = 4
+	// MultiJSON indicates the source is encoded in JSON-Array of individual records in Javscript Object Notation.
+	MultiJSON DataFormat = 5
+	// ORC indicates the source is encoded in Apache Optimized Row Columnar format.
+	ORC DataFormat = 6
+	// Parquet indicates the source is encoded in Apache Parquet format.
+	Parquet DataFormat = 7
+	// PSV is pipe "|" separated values.
+	PSV DataFormat = 8
+	// Raw is a text file that has only a single string value.
+	Raw DataFormat = 9
+	// SCSV is a file containing semicolon ";" separated values.
+	SCSV DataFormat = 10
+	// SOHSV is a file containing SOH-separated values(ASCII codepont 1).
+	SOHSV DataFormat = 11
+	// SingleJSON indicates the source containg a single Javascript Object Notation encoded record, newlines are treated as whitespace
+	SingleJSON DataFormat = 12
+	// SStream indicats the source is encoded as a Microsoft Cosmos Structured Streams format
+	SStream DataFormat = 13
+	// TSV is a file containing tab seperated values ("\t").
+	TSV DataFormat = 14
+	// TSVE is a file containing escaped-tab seperated values ("\t").
+	TSVE DataFormat = 15
+	// TXT is a text file with lines deliminated by "\n".
+	TXT DataFormat = 16
+	// WCLogFile indicates the source is encoded using W3C Extended Log File format
+	WCLogFile DataFormat = 17
+)
+
+type dfDescriptor struct {
+	camelName        string
+	jsonName         string
+	detectableExt    string
+	MappingReq       bool
+	validMappingKind bool
+}
+
+var dfDescriptions = []dfDescriptor{
+	{"", "", "", false, false},
+	{"Avro", "avro", ".avro", true, true},
+	{"ApacheAvro", "avro", "", true, false},
+	{"Csv", "csv", ".csv", false, true},
+	{"Json", "json", ".json", true, true},
+	{"MultiJson", "json", "", true, false},
+	{"Orc", "orc", ".orc", false, true},
+	{"Parquet", "parquet", ".parquet", false, true},
+	{"Psv", "psv", ".psv", false, false},
+	{"Raw", "raw", ".raw", false, false},
+	{"Scsv", "scsv", ".scsv", false, false},
+	{"Sohsv", "sohsv", ".sohsv", false, false},
+	{"SingleJson", "json", "", true, false},
+	{"SStream", "sstream", ".sstream", false, false},
+	{"Tsv", "tsv", ".tsv", false, false},
+	{"Tsve", "tsve", ".tsve", false, false},
+	{"Txt", "txt", ".txt", false, false},
+	{"EcLogFile", "eclogfile", ".eclogfile", false, false},
+}
+
 // String implements fmt.Stringer.
 func (d DataFormat) String() string {
-	ext, err := dfToExt(d)
+	ext, err := d.jsonName()
```

Comment:
Unless .jsonName() is getting used somewhere else where you are doing error detection, collapse jsonName() into String()